psutil/tests/test_linux.py

```diff
@@ -1450,6 +1450,25 @@ def test_pid_exists_no_proc_status(self):
             assert psutil.pid_exists(os.getpid())
             assert m.called
 
+    def test_get_proc_inodes_fd_broken(self):
+        # Simulate a case where /proc/{pid}/fd/{fd} symlink
+        # points to a file with full path longer than PATH_MAX
+
+        def get_inodes():
+            connections = psutil._pslinux.Connections()
+            connections._procfs_path = psutil._common.get_procfs_path()
+            return connections.get_proc_inodes(os.getpid())
+
+        p = psutil.Process()
+        files = p.open_files()
+        with open(self.get_testfn(), 'w'):
+            # give the kernel some time to see the new file
+            call_until(p.open_files, "len(ret) != %i" % len(files))
+            patch_point = 'psutil._pslinux.os.readlink'
+            with mock.patch(patch_point,
+                            side_effect=OSError(errno.ENAMETOOLONG, "")) as m:
+                assert not get_inodes()
+                assert m.called
```

Comment:
It looks like in here you mock `Connections` class but you're testing `Process.open_files()` instead of `Process.connections()`.